frame/democracy/src/lib.rs

```diff
@@ -0,0 +1,2084 @@
+	#[pallet::storage]
+	#[pallet::getter(fn referendum_info)]
+	pub type ReferendumInfoOf<T: Config> = StorageMap<
+		_,
+		Twox64Concat,
+		ReferendumIndex,
+		ReferendumInfo<T::BlockNumber, T::Hash, BalanceOf<T>, T::AssetId>,
+	>;
+
+	/// All votes for a particular voter. We store the balance for the number of votes that we
+	/// have recorded. The second item is the total amount of delegations, that will be added.
+	///
+	/// TWOX-NOTE: SAFE as `AccountId`s are crypto hashes anyway.
+	#[pallet::storage]
+	pub type VotingOf<T: Config> = StorageMap<
+		_,
+		Twox64Concat,
+		(T::AccountId, T::AssetId),
+		Voting<BalanceOf<T>, T::AccountId, T::BlockNumber>,
+		ValueQuery,
+	>;
+
+	/// Accounts for which there are locks in action which may be removed at some point in the
+	/// future. The value is the block number at which the lock expires and may be removed.
+	///
+	/// TWOX-NOTE: OK ― `AccountId` is a secure hash.
+	#[pallet::storage]
+	#[pallet::getter(fn locks)]
+	pub type Locks<T: Config> = StorageMap<_, Twox64Concat, T::AccountId, T::BlockNumber>;
+
+	/// True if the last referendum tabled was submitted externally. False if it was a public
+	/// proposal.
+	// TODO: There should be any number of tabling origins, not just public and "external"
+	// (council). https://github.com/paritytech/substrate/issues/5322
+	#[pallet::storage]
+	pub type LastTabledWasExternal<T> = StorageValue<_, bool, ValueQuery>;
+
+	/// The referendum to be tabled whenever it would be valid to table an external proposal.
+	/// This happens when a referendum needs to be tabled and one of two conditions are met:
+	/// - `LastTabledWasExternal` is `false`; or
+	/// - `PublicProps` is empty.
+	#[pallet::storage]
+	pub type NextExternal<T: Config> =
+		StorageValue<_, (ProposalId<T::Hash, T::AssetId>, VoteThreshold)>;
+
+	/// A record of who vetoed what. Maps proposal hash to a possible existent block number
+	/// (until when it may not be resubmitted) and who vetoed it.
+	#[pallet::storage]
+	pub type Blacklist<T: Config> = StorageMap<
+		_,
+		Identity,
+		ProposalId<T::Hash, T::AssetId>,
+		(T::BlockNumber, Vec<T::AccountId>),
+	>;
+
+	/// Record of all proposals that have been subject to emergency cancellation.
+	#[pallet::storage]
+	pub type Cancellations<T: Config> =
+		StorageMap<_, Identity, ProposalId<T::Hash, T::AssetId>, bool, ValueQuery>;
+
+	/// Storage version of the pallet.
+	///
+	/// New networks start with last version.
+	#[pallet::storage]
+	pub(crate) type StorageVersion<T> = StorageValue<_, Releases>;
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T: Config> {
+		_phantom: sp_std::marker::PhantomData<T>,
+	}
+
+	#[cfg(feature = "std")]
+	impl<T: Config> Default for GenesisConfig<T> {
+		fn default() -> Self {
+			GenesisConfig { _phantom: Default::default() }
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+		fn build(&self) {
+			PublicPropCount::<T>::put(0_u32);
```

Comment:
seems like send PR to original code to make it more typed like:)

Reply:
I think clippy reported error on this. Will see if we can keep it to original

Reply:
Skipped this clippy check